web/components/input/buttons/ProgressGroup/ProgressGroup.test.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { mount, configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+import ProgressGroup from '.';
+
+beforeAll(() => {
+  configure({ adapter: new Adapter() });
+});
+
+describe('ProgressGroup component', () => {
+  let clicked;
+  let wrapper;
+
+  // Deep mount children of ProgressGroup
+  const getWrapper = props => mount(<ProgressGroup {...props} />);
+
+  describe('when all step are enabled', () => {
+    const count = 3;
+    const activeIndex = 1;
+    beforeEach(() => {
+      clicked = undefined;
+      wrapper = getWrapper({
+        count,
+        onClick: (i) => { clicked = i; },
+        activeIndex: 1,
+        lastValidIndex: count,
+      });
+    });
+
+    test('all buttons should be instantiated', () => {
+      for (let i = 0; i < count; i += 1) {
+        expect(wrapper.find('button').at(i).text()).toEqual((i + 1).toString());
+      }
+    });
+
+    test('all buttons should be clickable', () => {
+      for (let i = 0; i < count; i += 1) {
+        wrapper.find('button').at(i).simulate('click');
+        expect(clicked).toEqual(i);
+      }
+    });
+
+    test('active button should have class "active"', () => {
+      for (let i = 0; i < count; i += 1) {
+        if (i === activeIndex) {
+          expect(wrapper.find('button').at(i).hasClass('active')).toBeTruthy();
+        } else {
+          expect(wrapper.find('button').at(i).hasClass('active')).toBeFalsy();
+        }
+      }
+    });
+  });
+
+  describe('when some steps are disabled', () => {
+    const count = 6;
+    const activeIndex = 1;
+    const lastValidIndex = count / 2;
+
+    beforeEach(() => {
+      clicked = undefined;
+      wrapper = getWrapper({
+        count,
+        onClick: (i) => { clicked = i; },
+        activeIndex,
+        lastValidIndex,
+      });
+    });
+
+    test('all buttons should be instantiated', () => {
+      for (let i = 0; i < count; i += 1) {
+        expect(wrapper.find('button').at(i).text()).toEqual((i + 1).toString());
+      }
+    });
+
+    test('some buttons should not be clickable', () => {
+      for (let i = 0; i < count; i += 1) {
+        wrapper.find('button').at(i).simulate('click');
+        if (i <= lastValidIndex) {
+          expect(clicked).toEqual(i);
```

Comment:
can we test that the prop does not have disabled property when it's less than lastActiveIndex?

i.e. `expect(wrapper.childAt(i).props().disabled).toBeFalsy();`

I think it's especially important here since enzyme click simulations don't work with `disabled` attribute - `onClick` can be triggered even if the button is `disabled`, so testing that `onClick` is triggered by itself would not guarantee that the button is not `disabled`

Reply:
true - see fa0dd52